src/ansys/tools/versioning/utils.py

```diff
@@ -122,12 +122,17 @@ def version_tuple_as_string(version_tuple):
     """
     try:
         # Check version numbers are positive integers
-        if not all(isinstance(num, int) and num >= 0 for num in version_tuple):
+        if not all(
+            (isinstance(num, int) and num >= 0)
+            or (isinstance(num, str) and "dev" in num and len(num) <= 6)
```

Comment:
The number "6" seems pretty arbitrary to me... Why are we limiting it to 6?

Reply:
It is ... I just felt we should limitate the length of the tags

Reply:
Fine by me, though if we are not limiting the other version numbers length it feels awkward that we are limiting the one for dev 😄 

And there is another thing... right now, this code would work for:

```bash
123.123.dev912
123.dev231.0
dev123.523.123
```

``dev`` should only be accepted in the last element of the tuple, right?

And even this would be accepted:
```bash
123.123.devmycar
123.123.123devmycar645
...
```

IMO we should force it to only be at the beginning. And check that if we strip "dev" from the version, the rest of the elements are numbers.

Reply:
> dev should only be accepted in the last element of the tuple, right?

This will require to create a sort of new object which encompass the three version digits. I know it doesn't make sense, but I would leave it as it is for now?

Anyone how is doing versioning already should now that `dev` must be only on the last digit.

Reply:
Sure yeah, we can open a follow up issue/PR for this.